src/main/kotlin/stringcalculator/StringCalculator.kt

```diff
@@ -22,27 +21,18 @@ class StringCalculator {
         }
     }
 
-    private fun splitByCustomDelimiter(matchResult: MatchResult): List<String> {
+    private fun splitByCustomDelimiter(matchResult: MatchResult): Numbers {
         val customDelimiter = matchResult.groupValues[1]
         val splitTarget = matchResult.groupValues[2]
 
-        return splitTarget.split(customDelimiter)
+        return Numbers.from(splitTarget.split(customDelimiter))
     }
 
-    private fun splitByDefaultDelimiter(input: String) =
-        input.split(Regex(DEFAULT_DELIMITER_REGEX))
-
-    private fun validate(splitInputs: List<String>) {
-        splitInputs.forEach {
-            require(it.toIntOrNull() != null && it.toInt() >= 0) { "문자열 계산기에는 0, 양수만 인자로 올 수 있습니다!" }
-        }
-    }
-
-    private fun add(splitInputs: List<String>) =
-        splitInputs.fold(0, { accu, curr -> accu + curr.toInt() })
+    private fun splitByDefaultDelimiter(input: String): Numbers =
+        Numbers.from(input.split(DEFAULT_DELIMITER_REGEX))
 
     companion object {
-        const val CUSTOM_DELIMITER_FIND_REGEX = "//(.)\\\\n(.*)"
-        const val DEFAULT_DELIMITER_REGEX = "[,:]"
+        private val CUSTOM_DELIMITER_FIND_REGEX = Regex("//(.)\\\\n(.*)")
```

Comment:
수많은 이스케이프가 필요한 문자열의 경우 3중 따옴표 문자열을 사용하면 더 깔끔하게 표현할 수 있습니다.

```suggestion
        private val CUSTOM_DELIMITER_FIND_REGEX = Regex("""//(.)\\n(.*)""")
```